src/pymor/algorithms/newton.py

```diff
@@ -200,13 +200,15 @@ def res(x):
         if error_measure == 'update':
             err_scale_factor = solution_norm
 
-        logger.info(f'it:{iteration} '
-                    f'norm:{solution_norm:.3e} '
-                    f'upd:{update_norm:.3e} '
-                    f'rel_upd:{update_norm / solution_norm:.3e} '
-                    f'res:{residual_norm:.3e} '
-                    f'red:{residual_norm / residual_norms[-2]:.3e} '
-                    f'tot_red:{residual_norm / residual_norms[0]:.3e}')
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=RuntimeWarning)
```

Comment:
Maybe add a comment, why warnings are filtered? Seems fishy at first sight.

Reply:
In the tests, it happens that `solution_norm` is exactly zero, and, e.g., `np.float64(1) / np.float64(0)` raises a `RuntimeWarning`. Ignoring these warnings was an easy fix. Would it be better to check explicitly if `solution_norm` is zero?

Reply:
I'm ok with the code, but there should be a comment in the code explaining why warnings are filtered.

Reply:
I added an explanation now.